src/items/items.module.ts

```diff
@@ -0,0 +1,11 @@
+// module : service와 controller를 NestJS에 등록한다
+import { Module } from '@nestjs/common';
+import { ItemsService } from './items.service';
+import { ItemsController } from './items.controller';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+@Module({
+  providers: [ItemsService, PrismaService],
```

Comment:
```suggestion
  providers: [ItemsService],
```
현재 prisma.module이 Global 데코레이터로 설정되어 있어서 타 모듈에서 PrismaService를 import 하거나 provider에 등록하지 않아도 사용할 수 있습니다.

그리고 다른 도메인의 service 파일을 사용하려고 하신다면 provider에 등록하는 것 보다 해당 service class를 provider로 사용하고 있는 모듈을 import해서 사용하시는 것이 좋습니다.
provider에 등록하면 인스턴스가 새로 생성되지만 모듈을 import해서 쓰면 인스턴스를 추가로 생성하지 않고 기존에 있던 인스턴스를 사용하기 때문에 메모리 부하가 적습니다.

이런걸 싱글톤(singleton) 패턴이라고 합니다.
참고: https://velog.io/@leejm_dev/nestjs-singleton

Reply:
provider:[] 에 추가하지말고 import:[] 에 추가해서 써야 인스턴스를 생성하지 않는군요

Reply:
Global 데코레이터로 설정된 경우 provider등록하지 않아도 사용가능한 점 뿐만이 아니라, 모듈 임포트 / provider 등록의 차이점에 대해서 추가적 설명까지 해주셔서 감사합니다! 많이 배웠습니다!🐥✨